### PR TITLE
Add adc methods

### PR DESCRIPTION
**Pull request by RaulTrombin**
Update the adc_read method,
the future idea is to have a general concept for Channels for pwm, adc and so on.

### CODE REVIEW

File: examples/cpp/main.cpp [hunk 1]
```diff
@@ -36,6 +36,9 @@ int main() {
   printf("Reading ADC Channels: 1 = %i, 2 = %i, 3 = %i, 4 = %i\n",
          adc.channel[0], adc.channel[1], adc.channel[2], adc.channel[3]);
 
+  printf("Data ADC Channels: 1 = %i\n", read_adc(adc_Channel::Ch1)
+    );
+
```
Comment (joaoantoniocardoso) [inline examples/cpp/main.cpp:41]: that's a weird formatting

Comment (patrickelectric) [reply examples/cpp/main.cpp:41]: Check https://github.com/bluerobotics/navigator-lib/pull/16 to avoid such problems

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Style needs to be fixed.

File: examples/cpp/main.cpp [hunk 2]
```diff
@@ -36,6 +36,8 @@ int main() {
   printf("Reading ADC Channels: 1 = %i, 2 = %i, 3 = %i, 4 = %i\n",
          adc.channel[0], adc.channel[1], adc.channel[2], adc.channel[3]);
 
+  printf("Data ADC Channels: 1 = %i\n", read_adc(adc_Channel::Ch1));
```
Comment (joaoantoniocardoso) [inline examples/cpp/main.cpp:39]: Also here, the meaning of the resulting value of these ADC calls is not clear and should be documented.

**Comment by RaulTrombin** [comment]
> Style needs to be fixed.

@patrickelectric did a formating using style llvm

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**